test/integration-test-main.sh

```diff
@@ -366,6 +366,44 @@ function test_multipart_copy {
     rm_test_file "${BIG_FILE}-copy"
 }
 
+function test_multipart_mix {
+    describe "Testing multi-part mix ..."
+
+    if [ `uname` = "Darwin" ]; then
+       cat /dev/null > $BIG_FILE
+    fi
+    dd if=/dev/urandom of="/tmp/${BIG_FILE}" bs=$BIG_FILE_LENGTH seek=0 count=1
+    dd if="/tmp/${BIG_FILE}" of="${BIG_FILE}" bs=$BIG_FILE_LENGTH seek=0 count=1
+
+    # modify directly(seek 7.5MB offset)
+    MODIFY_START_BLOCK=$((15*1024*1024/2/4))
+    echo -n "0123456789ABCDEF" | dd of="${BIG_FILE}" bs=4 count=4 seek=$MODIFY_START_BLOCK conv=notrunc
+    echo -n "0123456789ABCDEF" | dd of="/tmp/${BIG_FILE}" bs=4 count=4 seek=$MODIFY_START_BLOCK conv=notrunc
+
+    # Verify contents of file
+    echo "Comparing test file"
+    if ! cmp "/tmp/${BIG_FILE}" "${BIG_FILE}"
+    then
+       return 1
+    fi
+
+    # modify directly(over file end offset)
+    # it needs to be within 32MB for s3proxy's PUT.
```

Comment:
This is because an error occurred in s3proxy when 1 part (that is the part of the last upload) wrote over 32 MB.
https://github.com/gaul/s3proxy/blob/master/src/main/java/org/gaul/s3proxy/S3Proxy.java#L136
I thought that I was affected by this, but please tell me if I　misunterstood.
However, the s3fs test here is a check that writes to positions that exceed the file size, so 26 MB is fine.

Reply:
This is a bit confusing but v4 requests have a limit whether in AWS or S3Proxy.  The workaround is chunked encoding which sends the length and payload hash for each chunk.  s3fs should probably support this but it generally is not an issue since multipart uses a 10 MB size.

Reply:
Removed this comment.